modal/app.py

```diff
@@ -405,7 +406,7 @@ def registered_web_endpoints(self) -> List[str]:
         return self._web_endpoints
 
     def local_entrypoint(
-        self, _warn_parentheses_missing=None, *, name: Optional[str] = None
+        self, _warn_parentheses_missing: Optional[bool] = None, *, name: Optional[str] = None
```

Comment:
We're just doing a truthiness check with it, I think a protocol that implements `__bool__` would be the most accurate. Any thoughts? `Any` covers 99% of cases though.

Reply:
In that case, it's just `None` though as well? As we have `Optional` to supplement `bool`.

Reply:
@erikbern Just following up here. Happy to switch it, just want to make sure I'm understanding the use-case.

Reply:
I think this could go either way. If it's annotated as `Any`, type checkers won't complain about the case where users forget to call the decorator, it will only be a runtime error. With the `bool` annotation, it will also be a type-checking error. The downside is that the type checking message won't be super clear about what the problem is. It's unfortunate that Python doesn't have a better way to handle the distinction between decorators and decorator factories.

Reply:
Are you fine if I just add support for both cases then?

Reply:
Ah I see what you're saying, I think we could fix this enforced `_warn_parentheses_missing` as a key-word argument, and made the first argument an optional callable, then add overloads for both cases (called/verus not called)

Reply:
https://github.com/modal-labs/modal-client/pull/1875/commits/e03b4e54baab846cd9fe5f1c1db01e896e8e3966

Reply:
> enforced _warn_parentheses_missing as a key-word argument

Might not be following but I don't think we want to do this? It's not something a user should ever spell out, it just exists to catch the non-called-decorator case with a better runtime error message.